File: src/AHF-API/DataLayer/Model/Team.cs
﻿using DataLayer.Model.InventoryModel;
using System;
using System.Collections.Generic;
using System.Text;

namespace DataLayer.Model
{
    public class Team
    {
        public Team(string name, int money)
        {
            Name = name;
            Money = money;
            Players = new List<Player>();
            PreviousLocations = new List<PreviousLocation>();
        }
        public Team()
        {
            Players = new List<Player>();
            PreviousLocations = new List<PreviousLocation>();
        }
        public int Id { get; set; }
        public string Name { get; set; }
        public List<Player> Players { get; set; }
        public Inventory Inventory { get; set; }
        public int Money { get; set; }
        public int TimerOffset { get; set; }
        public string Blackout { get; set; }
        public List<PreviousLocation> PreviousLocations { get; set; }
    }
}

File: src/AHF-App/app/src/main/java/com/ahf/antwerphasfallen/Model/Team.java
package com.ahf.antwerphasfallen.Model;

import com.ahf.antwerphasfallen.Model.Inventory;
import com.ahf.antwerphasfallen.Model.Player;

import java.util.List;

/**
 * Created by Jorren on 20/10/2018.
 */

public class Team {
    private int id;
    private String name;
    private List<Player> players;
    private Inventory inventory;
    private int money;
    private int timerOffset;
    private String blackout;

    public Team() {
    }

    public Team(String name) {
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Player> getPlayers() {
        return players;
    }
    
    public Inventory getInventory() {
        return inventory;
    }

    public void setInventory(Inventory inventory) {
        this.inventory = inventory;
    }

    public int getMoney() { return money;}

    public String getBlackout() {
        return blackout;
    }

    public int getTimerOffset() {
        return timerOffset;
    }

    public void setTimerOffset(int timerOffset){
        this.timerOffset = timerOffset;
    }
}
